glue/sales_val_flagging.py

```diff
@@ -152,6 +144,63 @@ def group_size_adjustment(df, stat_groups: list, min_threshold, condos: bool):
         df: dataframe with newly manually adjusted outlier values.
 
     """
+
+    # Assign sv_outlier_reasons
+    df["sv_outlier_reason1"] = np.nan
+    df["sv_outlier_reason2"] = np.nan
+    df["sv_outlier_reason3"] = np.nan
+
+    outlier_type_crosswalk = {
+        "sv_ind_ptax_flag_w_deviation": "PTAX-203 Exclusion",
+        "sv_ind_price_high_price": "High price",
+        "sv_ind_price_low_price": "Low price",
+        "sv_ind_price_high_price_sqft": "High price per square foot",
+        "sv_ind_price_low_price_sqft": "Low price per square foot",
+        "sv_ind_char_short_term_owner": "Short-term owner",
+        "sv_ind_char_family_sale": "Family Sale",
+        "sv_ind_char_non_person_sale": "Non-person sale",
+        "sv_ind_char_statistical_anomaly": "Statistical Anomaly",
+        "sv_ind_char_price_swing_homeflip": "Price swing / Home flip",
+    }
+
+    def fill_outlier_reasons(row):
+        reasons = np.array(
+            [row[f"sv_outlier_reason{i}"] for i in range(1, 4)], dtype=object
+        )
+
+        for col in outlier_type_crosswalk:
+            if row[col]:  # If the condition is True
+                reason = col  # Use column name as reason
+                # Check if reason is not already recorded and there's a np.nan slot to fill
+                if reason not in reasons and pd.isna(reasons).any():
+                    next_index = pd.isna(
+                        reasons
+                    ).argmax()  # Find the index of the first True (NaN)
+                    row[f"sv_outlier_reason{next_index + 1}"] = reason
+                    reasons[next_index] = reason  # Update reasons array
+
+        return row
```

Comment:
**[Suggestion, non-blocking]** I think we could simplify this a bit based on the following assumptions:

1. All `sv_outlier_reason` columns will be NA at the start of this block, because we set them that way on lines 149-151
2. All keys in `outlier_type_crosswalk` are unique

If either of these two assumptions are false, let me know and disregard this suggestion; otherwise, we could try this:

```suggestion
    def fill_outlier_reasons(row):
        reason_idx = 1
        for col in outlier_type_crosswalk:
            if row[col]:  # If the condition is True
                reason = col  # Use column name as reason
                row[f"sv_outlier_reason{reason_idx}"] = reason
                if reason_idx >= 3:
                    break
                reason_idx += 1

        return row
```